### PR TITLE
Replace calender deprecated methods

### PR DESCRIPTION
**Pull request by tobiasjungmann**
Removes deprecated Methods `requestPermissions()` and `onActivityResults()` with a non deprecated equivalent

# How to test: 
1. Open calender view.
2. Three dots -> Export to Goolge Calender once with - accept permission - deny permission to test both options.
3. Three dots -> Remove from Google Calender - accept permission - deny permission.
4. Plus in Toolbar -> Add new calender entry -save - must not show an error message.

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarController.kt [hunk 1]
```diff
@@ -34,6 +34,7 @@ class CalendarController(private val context: Context) : ProvidesCard, ProvidesN
     private val roomLocationsDao: RoomLocationsDao = TcaDb.getInstance(context).roomLocationsDao()
     private val widgetsTimetableBlacklistDao: WidgetsTimetableBlacklistDao = TcaDb.getInstance(context).widgetsTimetableBlacklistDao()
     private val eventColorProvider: EventColorController = EventColorController(context)
+    var requestCode: Int=0
```
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarController.kt:37]:
```suggestion
    var requestCode: Int = 0
```

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarFragment.kt [hunk 2]
```diff
@@ -44,6 +47,7 @@ class CalendarFragment :
     ),
     CalendarDetailsFragment.OnEventInteractionListener {
 
+
```
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarFragment.kt:50]:
```suggestion
```

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarFragment.kt [hunk 3]
```diff
@@ -398,24 +411,23 @@ class CalendarFragment :
...
-            val failed = data?.getStringExtra("failed")
-            val sum = data?.getStringExtra("sum")
-            ThemedAlertDialogBuilder(requireContext())
-                .setTitle(R.string.error_something_wrong)
-                .setMessage(getString(R.string.create_event_some_failed, failed, sum))
-                .setPositiveButton(R.string.ok, null)
-                .show()
+    private val startForResult =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult())
+        { result: ActivityResult ->
```
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarFragment.kt:420]:
```suggestion
        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result: ActivityResult ->
```

**Review by CommanderStorm — COMMENTED** [review]
Formatting fixes

**Review by kordianbruck — APPROVED** [review]

**Pull request merged**